packages/Toast/src/Toast.js

```diff
@@ -208,7 +205,7 @@ const defaultProps = {
   autoCloseDelay: 5000,
   children: null,
   hasCloseButton: true,
-  isOpen: undefined,
+  isOpen: null,
```

Comment:
What are your reasons for preferring `null` to `undefined`?

Reply:
My understanding of the distinction is that `undefined` is a type and the intent is that the variable has been declared but hasn't been assigned a value. While `null` is an explicit assignment and **purposefully** having an 'empty' value/object.

I figured that when the consumer decides to use it as a `controlled` or `uncontrolled` component, the intent of the prop is clear. If isOpen is provided, we should have it declared and assigned to a value they control, or if it's not, we can **expect** that it is going to be empty or unknown, and not used at all in the component.

To me, using undefined seems to imply, that it might be defined and assigned a value later down the road in it's lifetime, which is not the case if you decide to use the Toast as an uncontrolled component. What do you think?

https://stackoverflow.com/questions/5076944/what-is-the-difference-between-null-and-undefined-in-javascript

Reply:
> using undefined seems to imply, that it might be defined and assigned a value later down the road
I'm not sure I would make that assumption, but at any rate I'm content that you've given it some deliberate thought.  

I actually don't think there is a _right_ answer here, but in some of our other components, a value of `null` is an expected value provided by the consumer, and has some meaning for the component, so this pattern of testing if `someProp === null` was not possible, and I've been forced to rely on `undefined` instead.  

For consistency across the library it might be worth keeping `isControlled = someProps !== undefined`. But this is not a convention that is documented, or has really been discussed, and changing it later would not be a disruptive change in this case, so I'm okay with leaving it as is.

Reply:
I actually prefer `undefined`, `React` by default marks a property undefined when is not being passed down, in this cases if `typeof isOpen === "undefined"` you know for sure that is uncontrolled

using the value null could also mean `false`

let's say you have something crazy like:
```
const MyComponent = ()=> {
  // should I render?
  if(something) return <Some />
  return null
} 

function App() {
  const myComponent = <Mycomponent somenthing={false} />

  <Toast isOpen={myComponent} />
}
```

which would lead to a silent bug because `null` will mark the component as uncontrolled. 

Whenever an action depends on the existence of a prop, I prefer to use undefined and avoid the temptation of initializing the value. 

But this is a preference I don't see entirely incorrectly to use `null` to be honest, but I do think that undefined is a little more bulletproof.

Reply:
point in case lol https://github.com/acl-services/paprika/pull/1087 internally the ListBox need to have three states for isOpen, `null, true, false` and the popover uses null to figure out if it's controlled or uncontrolled.

The ListBox has always initialized the Popover as uncontrolled and was tricky to figure out why the toggle wasn't working correctly :) ...

Reply:
I see, very insightful! TIL.